src/System.Management.Automation/engine/LanguagePrimitives.cs

```diff
@@ -2908,27 +2909,45 @@ private static bool TryScanNumber(string strToConvert, Type resultType, out obje
         {
             var strToConvert = valueToConvert as string;
             Diagnostics.Assert(strToConvert != null, "Value to convert must be a string");
-            Diagnostics.Assert(IsNumeric(GetTypeCode(resultType)), "Result type must be numeric");
+            Diagnostics.Assert(
+                IsNumeric(GetTypeCode(resultType)) || resultType == typeof(BigInteger),
+                "Result type must be numeric");
 
             if (strToConvert.Length == 0)
             {
                 typeConversion.WriteLine("Returning numeric zero.");
+
+                // BigInteger is not IConvertible and will throw from ChangeType.
```

Comment:
My ask https://github.com/dotnet/runtime/issues/36264

Reply:
I'm curious as to their response, but I think it's unlikely it'll change overall. If `BigInteger` gets to be `IConvertible`, sooner or later someone's gonna come along and ask if `System.Numerics.Complex` should be as well, and... let's just say I'm watching that debate from the sidelines. 😂 🍿